Products/DataCollector/zendisc.py

```diff
@@ -178,27 +186,31 @@ def discoverRanges(self):
         defer.returnValue(devices)
 
     @defer.inlineCallbacks
-    def discoverRouters(self, rootdev, seenips=None):
+    def discoverRouters(self, rootdev):
         """
         Discover all default routers based on DMD configuration.
 
         @param rootdev {device class} device root in DMD
-        @param seenips {list} list of IP addresses
         """
-        if not seenips:
-            seenips = []
-        ips = yield self.config().callRemote("followNextHopIps", rootdev.id)
-        for ip in ips:
-            if ip in seenips:
-                continue
-            self.log.info("Device '%s' next hop '%s'", rootdev.id, ip)
-            seenips.append(ip)
-            router = yield self.discoverDevice(
-                ip, devicepath="/Network/Router"
+        observedIps = set()
+        routers = [rootdev]
+        while routers:
+            router = routers.pop(0)
+            ips = yield self.config().callRemote(
+                "followNextHopIps", router.id
+            )
+        unknownIps = (ip for ip in ips if ip not in observedIps)
+        for batch_of_ips in chunk(unknownIps, self.options.parallel):
+            results = yield defer.DeferredList(
+                self.discoverDevice(ip, devicepath="/Network/Router")
+                for ip in batch_of_ips
+            )
+            observedIps.update(batch_of_ips)
+            routers.extend(
+                foundRouter
+                for _, foundRouter in results
+                if foundRouter is not None
             )
```

Comment:
This code looks incomplete.

I compared the changes in this PR to the actual code in the support/6.x branch and they do not match.

I think there are commits missing.